### PR TITLE
Conn w config gui

### PR DESCRIPTION
**Pull request by borismarin**
adding panel with large buttons which hopefully clarify each one of the layouts in the conn widg
image

@tarelli needs css'ing

### CODE REVIEW

File: src/main/webapp/js/widgets/connectivity/Connectivity.js [hunk 1]
```diff
@@ -289,9 +290,72 @@ define(function (require) {
...
+            }
+
+            for(layout in layoutOptions){
+                deck.append(createCard(layoutOptions[layout]));
+            }
+
+            return container;
+        },
+
+        configViaGUI : function() {
```
Comment (borismarin) [inline src/main/webapp/js/widgets/connectivity/Connectivity.js]: @tarelli discuss

File: src/main/webapp/js/widgets/connectivity/Connectivity.js [hunk 2]
```diff
@@ -289,9 +290,72 @@ define(function (require) {
...
+            var that = this;
+            var popup = G.addWidget(1).setMessage(this.createLayoutSelector()[0].outerHTML).setSize(422.8,667.8);
+            popup.showTitleBar(false);
+            popup.$('.card').on('click', function(event) {
+                var netTypes = GEPPETTO.ModelFactory.getAllTypesOfType(GEPPETTO.ModelFactory.geppettoModel.neuroml.network)
```
Comment (borismarin) [inline src/main/webapp/js/widgets/connectivity/Connectivity.js]: @tarelli discuss

**Comment by tarelli** [comment]
@borismarin @pgleeson I've integrated this, made some changes, used a modal instead of Popup widget for the configuration, changed the flow so that you can preview the different layouts as you click on them, added a way to configure the same widget with a new button in the title bar and some usability improvements here and there, hope you'll like it!

**Comment by pgleeson** [comment]
My local install is not happy with this line: https://github.com/openworm/org.geppetto.frontend/blob/development/src/main/webapp/js/widgets/connectivity/Connectivity.js#L317

```
[INFO] --- jsdoc3-maven-plugin:1.1.0:jsdoc3 (default) @ frontend ---
[INFO] Generating...
[INFO] js: "/home/padraig/geppetto/org.geppetto.frontend/src/main/webapp/js/widgets/connectivity/Connectivity.js", line 317: missing formal parameter
[INFO] js:             function dedent(callSite, ...args) {
[INFO] js: .......................................^
[INFO] js: "/home/padraig/geppetto/org.geppetto.frontend/src/main/webapp/js/widgets/connectivity/Connectivity.js", line 317: missing } after function body
[INFO] js:             function dedent(callSite, ...args) {
[INFO] js: .......................................^
[INFO] js: "/home/padraig/geppetto/org.geppetto.frontend/src/main/webapp/js/widgets/connectivity/Connectivity.js", line 318: missing ; before statement
[INFO] js:                 function format(str) {
[INFO] js: .....................................^
```

Some recent addition in that wonderful language JavaScript?

**Comment by tarelli** [comment]
Yep, looking at it with @borismarin

**Comment by tarelli** [comment]
@pgleeson try again pulling the latest!

**Comment by pgleeson** [comment]
Great. Works now. Thanks!

**Pull request merged**